tests/subsys/bootloader/external_flash/src/main.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2012-2014 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <app_update.h>
+#include <dfu/flash_img.h>
+#include <dfu/mcuboot.h>
+#include <drivers/gpio.h>
+
+struct k_work upgrade;
+
+void upgrade_work(struct k_work *item){
+	printk("Trying to request upgrade\n");
+	int err = boot_request_upgrade(BOOT_UPGRADE_TEST);
+	printk("upgrade requested: %d \n", err);
+	if (err != 0) {
+		__ASSERT(true, "reguest upgrade failed %d", err);
+	}
+	printk("Upgrade done\n");
+}
+
+void button_pressed(struct device *dev, struct gpio_callback *cb,
+		    u32_t pins)
+{
+	printk("Button pressed\n");
+	k_work_submit(&upgrade);
+
+}
+
+static struct gpio_callback button_cb_data;
+
+void main(void)
+{
+	k_work_init(&upgrade, upgrade_work);
+	struct device *dev_button;
+	int ret;
+
+	dev_button = device_get_binding(DT_ALIAS_SW0_GPIOS_CONTROLLER);
```

Comment:
Can you use dk_buttons_and_leds for the button handling here? I think it would be simpler.

Reply:
Yes probably better.